src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.cs

```diff
@@ -988,5 +981,38 @@ private enum CurrentOperationType
             Write,
             Flush
         }
+
+        private void ExtractPathAndPathBase(string originalRequestPath, int numSegmentsInVirtualPath, out string requestPathBase, out string requestPath)
```

Comment:
https://github.com/aspnet/HttpAbstractions/blob/87cd79d6fc54bb4abf07c1e380cd7a9498a78612/src/Microsoft.AspNetCore.Http.Abstractions/Extensions/UsePathBaseMiddleware.cs#L54-L59
In fact you may not need to do this here at all. You could do it the way IISIntegration out-of-proc does and just call UsePathBase once at startup.

Reply:
Yeah that may be the way to go. Let me try to get that working.